src/components/box/ProductItem/ProductItem.tsx

```diff
@@ -1,59 +1,38 @@
 import styled from '@emotion/styled';
 import { useState, useEffect } from 'react';
-import { useRecoilState } from 'recoil';
-import { CartIcon } from '../../assets';
-import type { CartItem, Product } from '../../types/types';
-import { Text } from '../common/Text/Text';
-import InputStepper from '../common/InputStepper/InputStepper';
-import { cartListState } from '../../service/atom';
-import { useLocalStorage } from '../../hooks/useLocalStorage';
-import getPriceFormat from '../../utils/getPriceFormat';
+import { CartIcon } from '../../../assets';
+import type { Product } from '../../../types/types';
+import { Text } from '../../common/Text/Text';
+import InputStepper from '../../common/InputStepper/InputStepper';
+import getPriceFormat from '../../../utils/getPriceFormat';
+import { useCart } from '../../../hooks/useCart';
 
 const ProductItem = ({ product }: { product: Product }) => {
-  const { localStorageData } = useLocalStorage<CartItem[]>('cartList', []);
+  const { data, addCartItemAPI, changeCartQuantityAPI, deleteCartItemAPI } = useCart();
+  const cartItemData = data && data.find((cart) => cart.product.id === product.id);
+  const cartId = cartItemData && cartItemData.id;
 
   const [quantity, setQuantity] = useState<number>(
-    localStorageData.find((data) => data.product.id === product.id)?.quantity ?? 0,
+    (data && data.find((data) => data.product.id === product.id)?.quantity) || 0,
```

Comment:
```suggestion
    (cartItemData?.quantity) ?? 0,
```

에디가 작성한 코드와 어떤 차이가 있을까요?

Reply:
훨씬 더 간단해졌네요! 그리고 ||을 사용하면 truthy값만 반환되니까 0은 결국 나오지 않겠네요..? ?? 가 맞는 연산자인 것 같습니다!